### PR TITLE
battle_bg

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: src/battle_bg.c [hunk 1]
```diff
@@ -0,0 +1,925 @@
...
+        break;
+    }
+}
+
+void DrawBattleEntryBackground(void)
+{
+    if (gBattleTypeFlags & BATTLE_TYPE_LINK)
+    {
+        LZDecompressVram(gFile_graphics_battle_transitions_vs_frame_sheet, (void*)(BG_CHAR_ADDR(1)));
+        LZDecompressVram(gFile_graphics_battle_transitions_vs_sheet, (void*)(VRAM + 0x10000));
```
Comment (jiangzhengwenjz) [inline src/battle_bg.c]: `gVsLettersGfx` in emerald

File: src/battle_bg.c [hunk 2]
```diff
@@ -0,0 +1,925 @@
...
+        LoadCompressedPalette(gFile_graphics_battle_transitions_vs_frame_palette, 0x60, 0x20);
+        SetBgAttribute(1, BG_ATTR_SCREENSIZE, 1);
+        SetGpuReg(REG_OFFSET_BG1CNT, 0x5C04);
```
Comment (jiangzhengwenjz) [inline src/battle_bg.c]: use constants

File: src/battle_bg.c [hunk 3]
```diff
@@ -0,0 +1,925 @@
...
+        CopyToBgTilemapBuffer(1, gFile_graphics_battle_transitions_vs_frame_tilemap, 0, 0);
+        CopyToBgTilemapBuffer(2, gFile_graphics_battle_transitions_vs_frame_tilemap, 0, 0);
+        CopyBgTilemapBufferToVram(1);
+        CopyBgTilemapBufferToVram(2);
+        SetGpuReg(REG_OFFSET_WININ, 0x36);
+        SetGpuReg(REG_OFFSET_WINOUT, 0x36);
```
Comment (jiangzhengwenjz) [inline src/battle_bg.c]: use constants

File: src/battle_bg.c [hunk 4]
```diff
@@ -0,0 +1,925 @@
...
+        gBattle_BG1_Y = 0xFF5C;
+        gBattle_BG2_Y = 0xFF5C;
```
Comment (jiangzhengwenjz) [inline src/battle_bg.c]: Negative value makes more sense here?

File: src/battle_bg.c [hunk 5]
```diff
@@ -0,0 +1,925 @@
...
+    gPlttBufferUnfaded[0x5C] = RGB( 9,  9,  9);
+    gPlttBufferUnfaded[0x5D] = RGB( 9,  9,  9);
+    gPlttBufferUnfaded[0x5E] = RGB(31, 31, 31);
+    gPlttBufferUnfaded[0x5F] = RGB( 26,  26,  25);
+    CpuCopy16(&gPlttBufferUnfaded[0x5C], &gPlttBufferFaded[0x5C], 8);
+    if (gBattleTypeFlags & (BATTLE_TYPE_FIRST_BATTLE | BATTLE_TYPE_POKEDUDE))
+    {
+        Menu_LoadStdPalAt(0x70);
+        TextWindow_LoadResourcesStdFrame0(0, 0x030, 0x70);
+        gPlttBufferUnfaded[0x76] = RGB( 0,  0,  0);
```
Comment (jiangzhengwenjz) [inline src/battle_bg.c]: `RGB_BLACK`

File: src/battle_bg.c [hunk 6]
```diff
@@ -0,0 +1,925 @@
...
+            BattlePutTextOnWindow(name, 0x10);
+
+            DrawLinkBattleParticipantPokeballs(taskId, playerId, 1, 2, 7);
+            DrawLinkBattleParticipantPokeballs(taskId, opponentId, 2, 2, 7);
+        }
+        gTasks[taskId].data[0]++;
+        break;
+    case 1:
+        palId = AllocSpritePalette(TAG_VS_LETTERS);
+        gPlttBufferUnfaded[palId * 16 + 0x10F] = gPlttBufferFaded[palId * 16 + 0x10F] = 0x7FFF;
```
Comment (jiangzhengwenjz) [inline src/battle_bg.c]: `RGB_WHITE`

File: src/battle_bg.c [hunk 7]
```diff
@@ -0,0 +1,925 @@
...
+        {
+            return 12;
+        }
+        else if (gTrainers[gTrainerBattleOpponent_A].trainerClass == CLASS_CHAMPION_2)
+        {
+            return 19;
+        }
+    }
+    battleScene = GetCurrentMapBattleScene();
+    if (battleScene == 0)
```
Comment (jiangzhengwenjz) [inline src/battle_bg.c]: `MAP_BATTLE_SCENE_NORMAL`

File: src/battle_bg.c (hunk 5 above)
Comment (PikalaxALT) [reply src/battle_bg.c]: Keeping it this way because surrounding RGB values are specified with parameters.

**Pull request merged**